code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -216,7 +216,7 @@
 	..()
 
 /datum/objective/revenant/check_completion()
-	if(!istype(owner.current, /mob/living/simple_animal/revenant) || !owner.current)
+	if(owner && !istype(owner.current, /mob/living/simple_animal/revenant) || !owner || !owner && !owner.current)
```

Comment:
Oh dear. This can simply be `!owner || !istype(owner.current, /mob/living/simple_animal/revenant)`